src/addrman.cpp

```diff
@@ -717,13 +717,17 @@ std::pair<CAddress, NodeSeconds> AddrManImpl::Select_(bool new_only, std::option
     size_t new_count = nNew;
     size_t tried_count = nTried;
 
-    if (network.has_value()) {
-        auto it = m_network_counts.find(*network);
-        if (it == m_network_counts.end()) return {};
+    if (networks.has_value()) {
+        new_count = 0;
+        tried_count = 0;
+        for (auto& network : *networks) {
+            auto it = m_network_counts.find(network);
+            if (it == m_network_counts.end()) return {};
```

Comment:
This should be `continue;`? If addrman has 0 addresses from some of the requested networks, it can still return an address from another requested network.

```suggestion
            if (it == m_network_counts.end()) {
                continue;
            }
```

Reply:
Yes, nice catch. I'll address it.